buildtest/buildsystem/base.py

```diff
@@ -274,23 +275,15 @@ def __str__(self):
     def __repr__(self):
         return self.__str__()
 
-    def _get_testdir(self):
-        """Based on the testfile path, return the testing directory.
-
-           Returns: full path to testing directory
-        """
-        return os.path.dirname(self.metadata["testpath"])
-
     def _create_test_folders(self):
         """Create all needed test folders on init, and add their paths
            to self.metadata.
         """
 
-        testdir = self._get_testdir()
-        create_dir(testdir)
+        create_dir(self.testdir)
```

Comment:
Didn't you remove this? This needs to be a function (property)

Reply:
Are you referring to the suggestion you mentioned in https://github.com/buildntest/buildtest/pull/283#discussion_r412489796 

Yes i removed ``self._get_testdir()`` it didn't make sense to me initially, If i understand you would like to retain this feature again?

Reply:
Instead of referencing self.metadata["testdir"] I suggested making a @property function so the user can reference `self.testdir`. I think I even wrote it out for you.

Reply:
Yes i got that, i think we have 2 identical ways to retrieve testdir and it's bit confusing why this was the case and i don't like it either. 

1. Class variable self.testdir on line 249 https://github.com/shahzebsiddiqui/buildtest-framework/blob/192fa17b61b96875ce8a491b903d60580ab01bae/buildtest/buildsystem/base.py#L249


2. self.metadata["testpath"], self.metadata["testdir"] on line 398 https://github.com/shahzebsiddiqui/buildtest-framework/blob/refactor_BuilderBase/buildtest/buildsystem/base.py#L398-399

It's bit redundant having self.testdir, self.metadata["testdir"] and i dont see a purpose of having both.

User can still pass a testdir to BuilderBase class and retrieve the test directory via class variable ``testdir``. So is adding a method ``testpath`` going to solve anything. I am fine btw, of referencing ``self.metadata['testpath']`` to keep things simple.

```
a = BuilderBase(name, recipe, testdir=/path/to/testdir)
a.testdir
```

I am not so familiar with @property, so i'd like to understand what is the end-goal with introducing this method.

Reply:
I prefer the first. When was the self.metadata ["testdir"] added? That doesn't make sense.

Reply:
https://www.freecodecamp.org/news/python-property-decorator/